### PR TITLE
chore: update default log level to info

### LINKED ISSUE #623: More documentation around LOG_LEVEL, set default to info
### Describe what should be investigated or refactored

We decided default log level should be info after adjusting the Admission and Webhook filters seeing to only fire based on correct binding labels, annotations, namespaces, etc. We are ignoring certain events from watch  but the logs are verbose in nature. Due to this, we want to make log level info by default.

AC:
- Log Level is info by default
- It is documented how to adjust log level
- we have verified the logs are less verbose 

### Additional context
Add any other context or screenshots about the technical debt here.

### PR DESCRIPTION
**Pull request by schaeferka**
## Description
Updated default LOG_LEVEL to info from debug.
...

## Related Issue

Fixes #623 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/pepr/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

**Comment by cmwylie19** [comment]
in monitor there is a console.debug call `console.debug(`\u001b[1;31m${filteredFailures}\u001b[0m`)`, we may want to change that to console.log so that it does not go away

File: src/lib/assets/yaml.ts [hunk 1]
```diff
@@ -77,7 +79,9 @@ export async function overridesFile({ hash, name, image, config, apiToken }: Ass
       env: [
         { name: "PEPR_WATCH_MODE", value: "true" },
         { name: "PEPR_PRETTY_LOG", value: "false" },
-        { name: "LOG_LEVEL", value: "debug" },
+        { name: "LOG_LEVEL", value: "info" },
+        process.env.PEPR_MODE === "dev" && { name: "MY_CUSTOM_VAR", value: "example-value" },
+        process.env.PEPR_MODE === "dev" && { name: "ZARF_VAR", value: "###ZARF_VAR_THING###" },
```
Comment (cmwylie19) [inline src/lib/assets/yaml.ts]:
```suggestion
        process.env.PEPR_MODE === "dev" && { name: "MY_CUSTOM_VAR", value: "example-value" },
        process.env.PEPR_MODE === "dev" && { name: "ZARF_VAR", value: "###ZARF_VAR_THING###" },
```

Can we please remove these lines? We cover this below this variable by pushing into the env array.

File: src/lib/assets/yaml.ts [hunk 2]
```diff
@@ -32,7 +32,9 @@ export async function overridesFile({ hash, name, image, config, apiToken }: Ass
       env: [
         { name: "PEPR_WATCH_MODE", value: "false" },
         { name: "PEPR_PRETTY_LOG", value: "false" },
-        { name: "LOG_LEVEL", value: "debug" },
+        { name: "LOG_LEVEL", value: "info" },
+        process.env.PEPR_MODE === "dev" && { name: "MY_CUSTOM_VAR", value: "example-value" },
+        process.env.PEPR_MODE === "dev" && { name: "ZARF_VAR", value: "###ZARF_VAR_THING###" },
```
Comment (cmwylie19) [inline src/lib/assets/yaml.ts]: Can we please remove these two lines? We cover this below.

**Review by cmwylie19 — DISMISSED** [review]
lgtm

File: src/lib/controller/index.ts [hunk 3]
```diff
@@ -235,14 +235,14 @@ export class Controller {
...
-          Log.debug({ ...reqMetadata, res }, "Check response");
+          Log.info({ ...reqMetadata, res }, "Check response");
         });
 
         let kubeAdmissionResponse: ValidateResponse[] | MutateResponse | ResponseItem;
 
         if (admissionKind === "Mutate") {
           kubeAdmissionResponse = response;
-          Log.debug({ ...reqMetadata, response }, "Outgoing response");
+          Log.info({ ...reqMetadata, response }, "Outgoing response");
```
Comment (cmwylie19) [inline src/lib/controller/index.ts]:
```suggestion
          Log.debug({ ...reqMetadata, response }, "Outgoing response");
```

We don't want to replace all the info with debug. We just want to preserve `npx pepr monitor` when the log level is info.

File: src/lib/controller/index.ts [hunk 4]
```diff
@@ -273,7 +273,7 @@ export class Controller {
           });
         }
 
-        Log.debug({ ...reqMetadata, kubeAdmissionResponse }, "Outgoing response");
+        Log.info({ ...reqMetadata, kubeAdmissionResponse }, "Outgoing response");
```
Comment (cmwylie19) [inline src/lib/controller/index.ts]:
```suggestion
        Log.debug({ ...reqMetadata, kubeAdmissionResponse }, "Outgoing response");
```

File: src/lib/controller/store.ts [hunk 5]
```diff
@@ -75,7 +75,7 @@ export class PeprControllerStore {
   };
 
   #receive = (store: PeprStore) => {
-    Log.debug(store, "Pepr Store update");
+    Log.info(store, "Pepr Store update");
```
Comment (cmwylie19) [inline src/lib/controller/store.ts]:
```suggestion
    Log.debug(store, "Pepr Store update");
```

File: src/lib/controller/store.ts [hunk 6]
```diff
@@ -182,7 +182,7 @@ export class PeprControllerStore {
     // Send any cached updates every debounceBackoff milliseconds
     setInterval(() => {
       if (Object.keys(sendCache).length > 0) {
-        Log.debug(sendCache, "Sending updates to Pepr store");
+        Log.info(sendCache, "Sending updates to Pepr store");
```
Comment (cmwylie19) [inline src/lib/controller/store.ts]:
```suggestion
        Log.debug(sendCache, "Sending updates to Pepr store");
```

File: src/lib/controller/store.ts [hunk 7]
```diff
@@ -192,7 +192,7 @@ export class PeprControllerStore {
 
   #createStoreResource = async (e: unknown) => {
     Log.info(`Pepr store not found, creating...`);
-    Log.debug(e);
+    Log.info(e);
```
Comment (cmwylie19) [inline src/lib/controller/store.ts]:
```suggestion
    Log.debug(e);
```

File: src/lib/controller/index.ts [hunk 8]
```diff
@@ -218,7 +218,7 @@ export class Controller {
         };
 
         Log.info({ ...reqMetadata, gvk, operation: request.operation, admissionKind }, "Incoming request");
-        Log.debug({ ...reqMetadata, request }, "Incoming request body");
+        Log.info({ ...reqMetadata, request }, "Incoming request body");
```
Comment (cmwylie19) [inline src/lib/controller/index.ts]:
```suggestion
        Log.debug({ ...reqMetadata, request }, "Incoming request body");
```

File: src/cli/dev.ts [hunk 9]
```diff
@@ -60,7 +60,7 @@ export default function (program: RootCmd) {
           program = fork(path, {
             env: {
               ...process.env,
-              LOG_LEVEL: "debug",
+              LOG_LEVEL: "info",
```
Comment (btlghrants) [inline src/cli/dev.ts]: Seeing this in code, I'm kinda feeling like `pepr dev` _should_ still be debug by default... :thinking: 

Given it's use during development, what do we think of keeping this one as verbose by default?

**Review by cmwylie19 — CHANGES_REQUESTED** [review]

File: src/cli/dev.ts (hunk 9 above)
Comment (cmwylie19) [reply src/cli/dev.ts]: I agree, during dev debug makes sense

**Review by cmwylie19 — CHANGES_REQUESTED** [review]

File: src/cli/dev.ts (hunk 9 above)
Comment (schaeferka) [reply src/cli/dev.ts]: Update made.

**Review by cmwylie19 — APPROVED** [review]

**Pull request merged**